SIPS/sip-26.md

```diff
@@ -0,0 +1,168 @@
+---
+sip: 26
+title: Account Router Snaps
+status: Draft
+author: Daniel Rocha (@danroc)
+created: 2024-08-28
+---
+
+## Abstract
+
+This SIP proposes a new API to be implemented by a new Account Router, allowing
+the forwarding of signing requests to the appropriate account Snap (i.e., Snaps
+that implement the [Keyring API][keyring-api]).
+
+## Motivation
+
+The Keyring API is being modified to support non-EVM chains. However, a
+challenge arises in identifying the correct account Snap that should receive
+the signing request, as this information is often only obtainable from the
+request itself, which varies based on the method and chain.
+
+## Specification
+
+> Formal specifications are written in TypeScript.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written
+in uppercase in this document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### High-Level architecture
+
+The diagram below represents a high-level architecture of how the Account
+Router integrates with the RPC Router, Account Snaps, and Protocol Snaps.
+
+![High-level architecture](../assets/sip-26/components-diagram.png)
+
+- **Account Snaps**: Snaps that implement the Keyring API and are responsible
+  for signing requests and managing accounts.
+
+- **Protocol Snaps**: Snaps that implement protocol methods that do not require
+  an account to be executed.
```

Comment:
I thought per our discussions with @rekmarks we'd determined that we can't enforce separation of Account + Protocol Snaps?
![Accounts SIP (1)](https://github.com/user-attachments/assets/6f7fe6a8-3a39-43e4-9773-814d052fe101)

Reply:
Oh I see the language saying these can be combined. I think its more confusing to represent them as (sortof) separate in the diagram. 

As I believe I said in our meeting I don't think we're really getting anything by having these non signing protocol snaps that implement non signing methods if the signing snaps can trivially handle those requests too.

In that case I feel like its clearer to call them something more general than "Account Snaps" since they are not just handling accounts. Maybe "Protocol Snaps" or "CAIP-2 Snaps" (though I'd prefer something that clearly states its purpose without external reference).

Reply:
To restate what I said in a meeting: As I read it, they are just two separate permissions that a Snap may or may not put together, and some Snaps (e.g. hardware wallet or SCAs) may want to only ship the account part.